### PR TITLE
[features-] add command to hide repetitive columns

### PR DESCRIPTION
**Pull request by midichef**
This is a command to hide any column that contains one value repeated in every row. Is this worth having? It's often the first command I run when opening large tables with such uninformative columns.

I'm not sure what to call the command.  Right now it is `hide-unvarying-cols`, but perhaps `hide-uniform-cols` is better? `hide-repetitive-cols` is too hard to spell.

### CODE REVIEW

**Comment by frosencrantz** [comment]
This would be a useful command.  Thank you for proposing it @midichef!

I like it and would use it.  Saul provided a useful suggestion when I asked in Issue 431 (2022 issues ago):

[[howto] hide invariant columns](https://github.com/saulpw/visidata/issues/431#top)

File: visidata/features/dedupe.py [hunk 1]
```diff
@@ -102,9 +102,21 @@ def _reload(self=vs):
...
+    for col in sheet.visibleCols:
+        vals = (col.getTypedValue(r) for r in sheet.rows)
+        first = next(vals)
+        if all(v == first for v in vals):
+            col.hide()
+
 # Add longname-commands to VisiData to execute these methods
 BaseSheet.addCommand(None, "select-duplicate-rows", "sheet.select_duplicate_rows()", "select each row that is a duplicate of a prior row")
 BaseSheet.addCommand(None, "dedupe-rows", "vd.push(sheet.dedupe_rows())", "open new sheet in which only non-duplicate rows in the active sheet are included")
+BaseSheet.addCommand(None, 'hide-unvarying-cols', 'sheet.hide_unvarying_cols()', 'hide any column that has multiple rows but only one distinct value')
```
Comment (saulpw) [inline visidata/features/dedupe.py]: These commands should all be on TableSheet instead of BaseSheet.

File: visidata/features/dedupe.py [hunk 2]
```diff
@@ -102,9 +102,21 @@ def _reload(self=vs):
...
 
 vd.addMenuItems('''
```
Comment (saulpw) [inline visidata/features/dedupe.py:109]: Let's add this command to an appropriate place in the menu.

Comment (midichef) [reply visidata/features/dedupe.py:109]: I think it'd be good to put it under `Column`, with `hide` and `unhide-cols`. I think the only way to do that is to do `addMenuItems` with the other hide commands in `features/layout.py`. Is that right? So then the function definition and `addCommand()` should move there.

File: visidata/features/dedupe.py (hunk 1 above)
Comment (midichef) [reply visidata/features/dedupe.py]: Okay. If I move the new command to `features/layout.py` (see below), `Sheet` is used for `Sheet.addCommand()` there, so I'll use that to match. What's the intended distinction between `Sheet` and `TableSheet`?

**Comment by midichef** [comment]
I changed the command prefix to `hide-uniform-` instead f `hide-unvarying-`. One reason is, "unvarying" is hardly ever used as a technical term, so it would be a new word for users to to recall. Another is, "hide" and "un-varying" describe the result based on two negatives (the positive equivalents would be "show" and "varying"). "uniform" seems better as it describes the targets of "hide" in terms of what they are (uniform), instead of what they aren't (varying).

If there's a common word that means "having one distinct value", I'd like to know.

File: visidata/features/dedupe.py (hunk 1 above)
Comment (saulpw) [reply visidata/features/dedupe.py]: There's no functional difference between Sheet and TableSheet.  Sheet is the original name, in use in many places and not worth it to fix.  TableSheet is the more descriptive name that is preferred for new usage.  But Sheet is fine to match, as you say.

**Comment by saulpw** [comment]
> I changed the command prefix to `hide-uniform-` instead f `hide-unvarying-`.

I was considering suggesting 'invariant' but I think I like `uniform` better (less formal).  Thanks for giving it some thought.

**Review by saulpw — APPROVED** [review]

File: visidata/features/dedupe.py (hunk 2 above)
Comment (saulpw) [reply visidata/features/dedupe.py:109]: Menu items can be added anywhere, so if this is the best place for the code, then the addCommand and its associated Menu item should be here.

**Pull request merged**